src/io/PImMSParser.m

```diff
@@ -0,0 +1,144 @@
+classdef PImMSParser < Parser
+    properties (Constant)
+        Name = 'PImMS Binary';
+    end
+
+    properties (SetAccess = protected)
+        % Time-of-flight properties
+        tof_start;
+        tof_end;
+        tof_range;
+
+        % Image data variable
+        image_data;
+    end
+
+    methods (Static)
+        function filterSpec = getFilterSpec()
+            filterSpec = {'*.bin', 'PImMS Binary (*.bin)'};
+        end
+    end
+
+    methods
+        function this = PImMSParser(filename)
+            this.filename = filename;
+
+            % Check that the filename ends in .bin
+            [pathstr,name,ext] = fileparts(filename);
+
+            if(~strcmpi(ext, '.bin'))
+                exception = MException('PImMSParser:FailedToParse', ...
+                    ['Must supply a BIN (*.bin) file']);
+                throw(exception);
+            end
+
+        end
+
+
+        function parse(this)
+            % Display a message to the user that parsing has started
+            notify(this, 'ParsingStarted');
+
+            % Assume data is PImMS2 (324x324)
+            % TODO: Perform any parsing of the data here. Making sure that width and height are filled in
+            this.width = 324;
+            this.height = 324;
+
+            % Set limits on how much of the time-of-flight to read
+            this.tof_start = 0;
+            this.tof_end = 4095;
```

Comment:
Similarly here, can these be pulled from the binary at all?

Reply:
Again, the start and end time bins containing signal are not explicitly included in the data. I think that this would definitely be something where it would be beneficial to have an additional loading GUI though, as you can then cut down the spectrum to just focus on the time-of-flight signal. I also hardcoded the maximum possible value of the timestamp in current lines 55 and 56, which should also be included in any GUI.